homeassistant/components/nibe_heatpump/climate.py

```diff
@@ -207,9 +214,12 @@ async def async_set_temperature(self, **kwargs: Any) -> None:
                     self._coil_setpoint_heat, temperature
                 )
             elif hvac_mode == HVACMode.COOL:
-                await coordinator.async_write_coil(
-                    self._coil_setpoint_cool, temperature
-                )
+                if self._coil_setpoint_cool:
+                    await coordinator.async_write_coil(
+                        self._coil_setpoint_cool, temperature
+                    )
+                else:
+                    raise ValueError(f"{hvac_mode} mode not supported for {self.name}")
```

Comment:
We should raise ServiceValidationError instead.

https://developers.home-assistant.io/docs/core/platform/raising_exceptions/

Reply:
I see. Then we should change line 222, 224 and 259 to rise ServiceValidationError.
And adapt the test. I do not see any other implication. 
@elupus do you see any side effects? 
I can quickly create the pr. Can I refer to the same ticket or do i need a new one?

Reply:
Go for it!

Reply:
there we go stack free error logs ;)
https://github.com/home-assistant/core/pull/117171